src/towncrier/_settings/load.py

```diff
@@ -43,10 +73,10 @@ def load_config_from_options(
         config = load_config(base_directory)
     else:
         config_path = os.path.abspath(config_path)
-        if directory:
-            base_directory = os.path.abspath(directory)
-        else:
+        if directory is None:
             base_directory = os.path.dirname(config_path)
+        else:
+            base_directory = os.path.abspath(directory)
```

Comment:
I've flipped this logic to mirror the logic from above, because it was confusing to have the same check inverted.